### ISSUE TITLE
SlotItemHandler#isSameInventory() regression

### ISSUE THREAD

**Issue by thecodewarrior** [MinecraftForge/MinecraftForge#7581]
**Minecraft Version:** 1.15.2, though the code hasn't changed in 1.16

**Forge Version:** 31.2.45

**Description of issue:**

A simple regression of #2884, caused by #5139 commenting out the [`SlotItemHandler#isSameInventory` override.](https://github.com/MinecraftForge/MinecraftForge/blob/1.16.x/src/main/java/net/minecraftforge/items/SlotItemHandler.java#L124)

Because all SlotItemHandlers use the same dummy inventory, without this override double-shift-clicking a `SlotItemHandler` slot will try to shift-click every item in every other `SlotItemHandler` slot, instead of only the items from the same `IItemHandler`.

![SlotItemHandler behavior](https://user-images.githubusercontent.com/5467669/103466669-c628d880-4cfb-11eb-9769-3b7c8ce60e68.gif)

**Comment by TheCurle** [comment]
Do you have a minimal test case? 
https://github.com/TheCurle/MinecraftForge/tree/1.16-slot-inventory 
I have a PR ready to go, but no easy way to test this

**Comment by thecodewarrior** [comment]
No, sorry. It's been four months since I opened this issue and I just started porting my library over to Quilt so I'm not particularly interested in setting up an environment to make one.
